### PR TITLE
Feature 3 new error handling

### PR DESCRIPTION
**Pull request by JuusoMRusanen**
Added an error handler wrapper using Jotai, and a generic dialog for it. 
Note: merging this will add Jotai state manager to this project.

### CODE REVIEW

File: src/components/generics/generic-dialog.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  onConfirm,
+  error,
+  fullScreen,
+  fullWidth,
+  maxWidth,
+  disableEnforceFocus,
+  disabled,
+  ignoreOutsideClicks,
+  children
+}: Props) => {
```
Comment (petri-nykanen) [inline src/components/generics/generic-dialog.tsx:55]: Dialog component has a list of props, some which are unused. For me this is fine, but if you want to more concise code then maybe the unnecessary props could be dropped for now.

File: src/App.tsx [hunk 2]
```diff
@@ -23,19 +24,21 @@ function App () {
...
-        userProfile={userProfile}
-        setUserProfile={setUserProfile}>
-        <RouterProvider router={router} />
-      </AuthenticationProvider>
+      <ErrorHandler>
+        <AuthenticationProvider
+          auth={auth}
+          setAuth={setAuth}
+          userProfile={userProfile}
+          setUserProfile={setUserProfile}>
```
Comment (Jdallos) [inline src/App.tsx:38]: If jotai is installed then shouldn't these props be accessed as atoms within the AuthenticationProvider

File: src/components/contexts/error-handler.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+const ErrorHandler = ({ children }: Props) => {
+  const [error, setError] = useAtom(errorAtom);
+
+  /**
+   * Handles error message
+   *
+   * @param message error message
+   * @param err any error
+   */
+  const handleError = async (message: string, err?: any) => {
```
Comment (Jdallos) [inline src/components/contexts/error-handler.tsx:28]: Can we type err with something more specific than any?

File: src/components/screens/error-screen.tsx [hunk 4]
```diff
@@ -5,7 +5,7 @@ import { useRouteError } from "react-router-dom";
  * 
  * @returns ReactElement
  */
-function ErrorPage () {
+function ErrorScreen () {
```
Comment (Jdallos) [inline src/components/screens/error-screen.tsx:8]: Convert to arrow function

**Review by Jdallos — APPROVED** [review]

File: src/components/generics/generic-dialog.tsx (hunk 1 above)
Comment (JuusoMRusanen) [reply src/components/generics/generic-dialog.tsx:55]: I think these are useful options to have in the generic dialog.

File: src/App.tsx (hunk 2 above)
Comment (JuusoMRusanen) [reply src/App.tsx:38]: Made the props accessible within the AuthenticationProvider using Jotai.

File: src/components/contexts/error-handler.tsx (hunk 3 above)
Comment (JuusoMRusanen) [reply src/components/contexts/error-handler.tsx:28]: Typed it as an object

File: src/components/screens/error-screen.tsx (hunk 4 above)
Comment (JuusoMRusanen) [reply src/components/screens/error-screen.tsx:8]: done

**Pull request merged**